src/main/java/dev/devature/penguin_api/service/EmailAuthService.java

```diff
@@ -0,0 +1,66 @@
+package dev.devature.penguin_api.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import dev.devature.penguin_api.interfaces.AuthenticationStrategy;
+import dev.devature.penguin_api.repository.AuthenticationRepository;
+import dev.devature.penguin_api.dto.AuthRequest;
+
+import java.util.regex.Pattern;
+
+@Service
+public class EmailAuthService implements AuthenticationStrategy {
+
+    private final String passwordRegex = "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[~`!@#$%^&*()\\-_+={}\\[\\]|;:<>,./?]).{8,}$";
+    private final String emailRegex = "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])";
```

Comment:
I think these regexes are also in the `RegisterService`? Rather than duplicate them, I would prefer they be static members of some class, and those referenced where they are needed like here and `RegisterService`

Reply:
I wasn't sure if I should change RegisterService, but if Tony's done with it I can put the regexes in a utility class.